hathor/indexes.py

```diff
@@ -37,64 +38,124 @@ class TransactionIndexElement(NamedTuple):
     hash: bytes
 
 
-class IndexesManager:
-    """ IndexesManager manages all the indexes that we will have in the system
+class IndexesManager(ABC):
```

Comment:
I think that this isn't the right class to be abstract. Since this isn't introducing any other implementation besides  `MemoryIndexesManager`, I'd prefer if this abstraction wasn't introduced right now, because I want to discuss a better approach of which classes to make abstract and how to construct them. Aside from that, everything else looks great to me.

Reply:
As we discussed, I'll move forward with this abstraction, and we will update it soon.